buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py

```diff
@@ -0,0 +1,167 @@
+from pynwb.behavior import SpatialSeries, Position, CompassDirection
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils.conversion_tools import get_module
+from nwb_conversion_tools.utils.json_schema import FolderPathType
+
+from .tingleyseptal_utils import read_matlab_file
+
+
+class TingleySeptalBehaviorInterface(BaseDataInterface):
+    """Behavior data interface for the Tingley Septal project."""
+
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        session_path = Path(self.source_data["folder_path"])
+        session_id = session_path.stem
+
+        # Load the file with behavioral data
+        behavior_file_path = Path(session_path) / f"{session_id}.behavior.mat"
+        behavior_mat = read_matlab_file(str(behavior_file_path))["behavior"]
+
+        # Add trials
+        events = behavior_mat["events"]
+        trial_interval_list = events["trialIntervals"]
+
+        data = []
+        for start_time, stop_time in trial_interval_list:
+            data.append(
+                dict(
+                    start_time=float(start_time),
+                    stop_time=float(stop_time),
+                )
+            )
+        [nwbfile.add_trial(**row) for row in sorted(data, key=lambda x: x["start_time"])]
+
+        trial_list = events["trials"]
+        direction_list = [trial.get("direction", "") for trial in trial_list]
+        trial_type_list = [trial.get("type", "") for trial in trial_list]
+
+        if not all([direction == "" for direction in direction_list]):
+            nwbfile.add_trial_column(name="direction", description="direction of the trial", data=direction_list)
+
+        if not all([trial_type == "" for trial_type in trial_type_list]):
+            nwbfile.add_trial_column(name="trial_type", description="type of trial", data=trial_type_list)
+
+        # Position
+        module_name = "Position"
+        module_description = "Contains behavioral data concerning position."
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        timestamps = np.array(behavior_mat["timestamps"])[..., 0]
+
+        position = behavior_mat["position"]
+        pos_data = [[x, y, z] for (x, y, z) in zip(position["x"], position["y"], position["y"])]
+        pos_data = np.array(pos_data)[..., 0]
+
+        units = behavior_mat.get("units", None)
+        if units == "m":
+            conversion = 1.0
+        else:
+            warnings.warn(f"Spatial units {units} not listed in meters; " "setting conversion to nan.")
+            conversion = np.nan
+
+        description = behavior_mat.get("description", "generic_position_tracking").replace("/", "-")
+        rotation_type = behavior_mat.get("rotationType", "non_specified")
+
+        pos_obj = Position(name=f"{description}")
```

Comment:
```suggestion
        pos_obj = Position(name=f"{description.replace(" ", "_")}")
```
Noticed that some of the original names have spaces in them.